File: services/Common/Cpm.Common.Query/Models/BaseSearchModel.cs
namespace Cpm.Common.Query.Models;

public record BaseSearchModel(
    IEnumerable<SearchCriterion> SearchCriteria,
    IEnumerable<SortCriterion> SortCriteria,
    int PageIndex,
    int PageSize
)
{
    public BaseSearchModel() : this(Array.Empty<SearchCriterion>(), Array.Empty<SortCriterion>(), 1, 10)
    {
    }
}

File: ui/src/domain/models/common/search/BaseSearchModel.ts
import { SortCriterion } from '@/domain/models/common/search/SortCriterion';
import { SearchCriterion } from '@/domain/models/common/search/SearchCriterion';

export interface BaseSearchModel {
  searchCriteria: SearchCriterion[];
  sortCriteria: SortCriterion[];
  pageIndex: number;
  pageSize: number;
}

export const defaultBaseSearchModel: BaseSearchModel = {
  searchCriteria: [],
  sortCriteria: [],
  pageIndex: 1,
  pageSize: 10,
};